### PR TITLE
Replace Op strings with Op types in decompose.py

### LINKED ISSUE #2875: [BUG] `DecompRuleInterpreter` performs string check rather than type check during `cleanup` method
Because of this hard string match [here](https://github.com/PennyLaneAI/catalyst/blob/main/frontend/catalyst/from_plxpr/decompose.py#L244) for "Controlled", this MWE raises an unexpected error,

```python 
import pennylane as qml
from pennylane.decomposition import controlled_resource_rep

qml.decomposition.enable_graph()


def _resources():
    return {
        controlled_resource_rep(qml.BasisEmbedding, {"num_wires": 1}, num_control_wires=1): 1,
        qml.resource_rep(qml.GlobalPhase): 1,
    }


@qml.register_resources(_resources)
def my_rz(phi, wires):
    qml.GlobalPhase(phi / 2)
    qml.ctrl(qml.BasisEmbedding([1], wires=[1]), control=wires)


@qml.qjit(capture=True)
@qml.decompose(gate_set={"CNOT", "PauliX", "GlobalPhase", "MultiControlledX"},
               fixed_decomps={qml.RZ: my_rz})
@qml.qnode(qml.device("null.qubit", wires=2))
def f(phi):
    qml.RZ(phi, 0)
    return qml.state()
```

```pycon
>>> f(0.5)
ValueError: Could not capture _OperatorNode(op=Controlled(BasisState(num_wires=1), num_control_wires=1, num_work_wires=0, num_zero_control_values=0, work_wire_type=borrowed), num_work_wire_not_available=0, work_wire_dependent=False, min_work_wires=0, reachable=True) without the number of wires.
```

### PR DESCRIPTION
**Pull request by maliasadi**
**Context:**
Replace Op strings with Op types in decompose.py

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**
[sc-120705]
Fixes #2875

### CODE REVIEW

File: frontend/catalyst/from_plxpr/decompose.py
Comment (astralcai) [inline frontend/catalyst/from_plxpr/decompose.py]:
```suggestion
                raise ValueError(f"Could not capture {op.op} without the number of wires.")
```
`op` is an `_OperatorNode`, so the error message often turns out to be confusing.

File: frontend/catalyst/from_plxpr/decompose.py [hunk 1]
```diff
@@ -237,14 +237,17 @@ def cleanup(self):
...
+                    op_type,
+                    (
+                        qp.ops.Adjoint,
+                        qp.ops.Controlled,
+                        qp.ops.ChangeOpBasis,
+                        qp.ops.Prod,
+                        qp.TemporaryAND,
+                    ),
                 )
             ):  # pragma: no cover
```
Comment (andrijapau) [inline frontend/catalyst/from_plxpr/decompose.py]: Why do we have a pragma no cover here? 🤔

File: frontend/catalyst/from_plxpr/decompose.py
Comment (maliasadi) [reply frontend/catalyst/from_plxpr/decompose.py]: ah this is outdated ! let me fix in a followup push

Comment (maliasadi) [reply frontend/catalyst/from_plxpr/decompose.py]: Fixed [d2cd267](https://github.com/PennyLaneAI/catalyst/pull/2873/commits/d2cd2672de38630260d8a54407051f90e48f8c8e)

File: frontend/catalyst/from_plxpr/decompose.py (hunk 1 above)
Comment (maliasadi) [reply frontend/catalyst/from_plxpr/decompose.py]: Added [17f2d55](https://github.com/PennyLaneAI/catalyst/pull/2873/commits/17f2d55105b364a177b13c2ab539fe5d4e38fcec)

File: frontend/test/pytest/from_plxpr/test_decompose_transform.py [hunk 2]
```diff
@@ -1085,6 +1086,88 @@ def c():
...
+                controlled_resource_rep(
+                    qp.BasisEmbedding, {"num_wires": 1}, num_control_wires=1
+                ): 1,
+                qp.resource_rep(qp.GlobalPhase): 1,
+            }
+
+        @qp.register_resources(_resources)
+        def my_rz(phi, wires):
+            qp.GlobalPhase(phi / 2)
+            qp.ctrl(qp.BasisEmbedding, control=wires)([1], wires=[1])
```
Comment (maliasadi) [inline frontend/test/pytest/from_plxpr/test_decompose_transform.py:1155]: `qml.ctrl(qml.BasisEmbedding([1], wires=[1]), control=wires)`  now triggers a capture issue. But the issue related to the string-based checking in decmpose.py is fixed in both cases

**Review by astralcai — APPROVED** [review]

**Review by andrijapau — COMMENTED** [review]
My only concern is here is that we're technically missing coverage for `Adjoint`, `ChangeOpBasis`, `Prod` and `TemporaryAND`, no?

**Comment by maliasadi** [comment]
> My only concern is here is that we're technically missing coverage for `Adjoint`, `ChangeOpBasis`, `Prod` and `TemporaryAND`, no?

@andrijapau each edge case is tested by the `TestPlxPRDecomposition` suite (just a few: `test_adjoint`, `test_ctrl`, `test_template_qft`, `test_multi_qubits`). Removing any of these Ops will break tests in `from_plxpr`.

Note this is a *temporary* workaround introduced a few cycles ago for `Adjoint`/`Ctrl` and later expanded for other ops! We should not rely on it long-term and will completely revisit this class after the new Operator is developed.

**Review by andrijapau — APPROVED** [review]
